src/sentry/api/endpoints/organization_events_timeseries.py

```diff
@@ -0,0 +1,371 @@
+from collections.abc import Mapping
+from datetime import timedelta
+from typing import Any, NotRequired, TypedDict
+
+import sentry_sdk
+from rest_framework.exceptions import ParseError, ValidationError
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases import NoProjects, OrganizationEventsV2EndpointBase
+from sentry.api.endpoints.organization_events_stats import (
+    ALLOWED_EVENTS_STATS_REFERRERS,
+    METRICS_ENHANCED_REFERRERS,
+    SENTRY_BACKEND_REFERRERS,
+)
+from sentry.constants import MAX_TOP_EVENTS
+from sentry.models.organization import Organization
+from sentry.search.eap.types import SearchResolverConfig
+from sentry.search.events.types import SnubaParams
+from sentry.snuba import (
+    discover,
+    errors,
+    functions,
+    metrics_enhanced_performance,
+    metrics_performance,
+    ourlogs,
+    spans_eap,
+    spans_indexed,
+    spans_metrics,
+    spans_rpc,
+    transactions,
+    uptime_checks,
+)
+from sentry.snuba.query_sources import QuerySource
+from sentry.snuba.referrer import Referrer
+from sentry.snuba.utils import DATASET_LABELS
+from sentry.utils.snuba import SnubaTSResult
+
+TOP_EVENTS_DATASETS = {
+    discover,
+    functions,
+    metrics_performance,
+    metrics_enhanced_performance,
+    spans_indexed,
+    spans_metrics,
+    spans_eap,
+    errors,
+    transactions,
+}
+
+
+class StatsMeta(TypedDict):
+    dataset: str
+    start: float
+    end: float
+
+
+class Row(TypedDict):
+    timestamp: float
+    value: float
+
+
+class SeriesMeta(TypedDict):
+    order: NotRequired[int]
+    groupBy: NotRequired[list[str]]
```

Comment:
Suggestion: we could add key and value here instead of just values?
ie 
`groupBy: {'transaction': 'foo', 'status': 'bar'}`
instead of 
`groupBy: ['foo', 'bar']`

Reply:
Resolving in favour of discussion on notion